sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogSuite.scala

```diff
@@ -154,4 +155,26 @@ class HiveExternalCatalogSuite extends ExternalCatalogSuite {
     catalog.createTable(hiveTable, ignoreIfExists = false)
     assert(catalog.getTable("db1", "spark_29498").owner === owner)
   }
+
+  test("SPARK-30868 throw an exception if HiveClient#runSqlHive fails") {
+    // test add jars which not exists
+    val jarPath = "file:///tmp/not_exists.jar"
+    assertThrows[QueryExecutionException](externalCatalog.client.runSqlHive(s"ADD JAR $jarPath"))
```

Comment:
nit: to shorten the test we can add `val client = externalCatalog.client` at the beginning.